### ISSUE TITLE
Packages dsdp, scip_sdp

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#34749]
<div id="comment:0"></div>

http://www.opt.tu-darmstadt.de/scipsdp/#about

https://github.com/scipopt/SCIP-SDP

used to be LGPL license, changed to Apache 2.0 in Dec 2022

For full functionality, needs an SDP solver, one of:
- DSDP - https://repology.org/project/dsdp/versions -- added here as a Sage package
- SDPA - https://repology.org/project/sdpa/versions
- MOSEK (non-free)

Depends on #31329
Depends on #34835

CC:  @dimpase @vbraun

Component: **packages: optional**

Author: **Matthias Koeppe**

Branch/Commit: **[`544d123`](https://github.com/sagemath/sagetrac-mirror/commit/544d123ab56ce2fee80e72c4b4ffeb1fabf305c9)**

Reviewer: **Dima Pasechnik**

_Issue created by migration from https://trac.sagemath.org/ticket/34749_

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,3 +1,3 @@
 http://www.opt.tu-darmstadt.de/scipsdp/#about
 
-LGPL license
+used to be LGPL license, changed to Apache 2.0 in Dec 2022
``````

**Comment by mkoeppe** [comment]
Dependencies: **#31329**

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,3 +1,11 @@
 http://www.opt.tu-darmstadt.de/scipsdp/#about
 
+https://github.com/scipopt/SCIP-SDP
+
 used to be LGPL license, changed to Apache 2.0 in Dec 2022
+
+For full functionality, needs an SDP solver, one of:
+- DSDP 
+- SDPA
+- MOSEK (non-free)
+
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -5,7 +5,8 @@
 used to be LGPL license, changed to Apache 2.0 in Dec 2022
 
 For full functionality, needs an SDP solver, one of:
-- DSDP 
-- SDPA
+- DSDP - https://repology.org/project/dsdp/versions
+- SDPA - https://repology.org/project/sdpa/versions
 - MOSEK (non-free)
 
+
``````

**Comment by mkoeppe** [comment]
Branch: **[u/mkoeppe/package_scip_sdp](https://github.com/sagemath/sagetrac-mirror/tree/u/mkoeppe/package_scip_sdp)**

**Comment by mkoeppe** [comment]
Commit: **[`8a954ab`](https://github.com/sagemath/sagetrac-mirror/commit/8a954ab20564d5d7e943d7e5a5f09d81b729391f)**

**Comment by mkoeppe** [comment]
<div id="comment:5" align="right">comment:5</div>

First draft of the install script

```
[scip_sdp-4.1.0] -- Build shared libraries: ON
[scip_sdp-4.1.0] -- Found lapack library: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/liblapack.tbd
[scip_sdp-4.1.0] -- Found blas library: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/libblas.tbd
```
needs to be changed so it uses our openblas

```
[scip_sdp-4.1.0] -- Finding symmetry computation program "bliss"
[scip_sdp-4.1.0] -- Finding BLISS
[scip_sdp-4.1.0] -- Finding BLISS - found
...
[scip_sdp-4.1.0] /Users/mkoeppe/s/sage/sage-rebasing/local/var/tmp/sage/build/scip_sdp-4.1.0/src/src/symmetry/compute_symmetry_bliss.cpp:42:10: fatal error: 'bliss/defs.hh' file not found
[scip_sdp-4.1.0] #include <bliss/defs.hh>
[scip_sdp-4.1.0]          ^~~~~~~~~~~~~~~
```
needs to be find scip's vendored bliss (or we need to unvendor - #33010/#34747)

---
Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/68fb9a7cb229f23bbbd94b3ead9747006ae243ae"><code>68fb9a7</code></a></td><td><code>build/pkgs/scipoptsuite/dependencies: Add papilo</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2dd029e6607e7ef6966dee83d5db5f92fd3121ed"><code>2dd029e</code></a></td><td><code>build/pkgs/{scipoptsuite,soplex}: Use papilo</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/5efb7ba56ff1050aaa339909473de3ebd978b57b"><code>5efb7ba</code></a></td><td><code>build/pkgs/soplex/dependencies: Add papilo</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0fc60a0c9a77ce58d1da4d017b5524e8553e65ec"><code>0fc60a0</code></a></td><td><code>build/pkgs/scipoptsuite: Fix install_name on macOS</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/392455dc743a3c17a6ba750c4f9354cb15701b7f"><code>392455d</code></a></td><td><code>build/pkgs/scip: Rename from scipoptsuite</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/45f957e011c946a26354615c4d86441c699a3617"><code>45f957e</code></a></td><td><code>Revert "build/pkgs/boost_cropped/distros/debian.txt: Add libboost-program-options-dev"</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/931e348433c3a0a3cf75423e3d50ada517f839b2"><code>931e348</code></a></td><td><code>Merge #34726</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/9b402723dff9a32c66ed4d5fbccacd586c6720cf"><code>9b40272</code></a></td><td><code>build/pkgs/scip/distros/conda.txt: New</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/62211802bfceef69f683328e81b76a2eaa49acd5"><code>6221180</code></a></td><td><code>Merge #31329</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/8a954ab20564d5d7e943d7e5a5f09d81b729391f"><code>8a954ab</code></a></td><td><code>build/pkgs/scip_sdp: New</code></td></tr></table>

**Comment by mkoeppe** [comment]
<div id="comment:6" align="right">comment:6</div>

The failure with bliss was due to a broken local installation

**Comment by mkoeppe** [comment]
<div id="comment:7" align="right">comment:7</div>

scip_sdp should probably use https://cmake.org/cmake/help/latest/module/FindLAPACK.html, https://cmake.org/cmake/help/latest/module/FindBLAS.html

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`8a954ab`](https://github.com/sagemath/sagetrac-mirror/commit/8a954ab20564d5d7e943d7e5a5f09d81b729391f)** to **[`8250adb`](https://github.com/sagemath/sagetrac-mirror/commit/8250adb72631c18d7db630b102c715bde64a5d7a)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:8"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/8250adb72631c18d7db630b102c715bde64a5d7a"><code>8250adb</code></a></td><td><code>build/pkgs/scip_sdp/spkg-install.in: Better config</code></td></tr></table>

**Comment by mkoeppe** [comment]
<div id="comment:9" align="right">comment:9</div>

Many linker errors on macOS:

```
[scip_sdp-4.1.0] [100%] Linking CXX shared library ../lib/libscipsdp.dylib
[scip_sdp-4.1.0] cd /Users/mkoeppe/s/sage/sage-rebasing/local/var/tmp/sage/build/scip_sdp-4.1.0/src/build/src && /usr/local/Cellar/cmake/3.24.2/bin/cmake -E cmake_link_script CMakeFiles/libscipsdp.dir/link.txt --verbose=1
[scip_sdp-4.1.0] /Library/Developer/CommandLineTools/usr/bin/g++  -std=gnu++11 -g -O2 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk -dynamiclib -Wl,-headerpad_max_install_names -L/Users/mkoeppe/s/sage/sage-rebasing/local/lib -Wl,-rpath,/Users/mkoeppe/s/sage/sage-rebasing/local/lib -L/Users/mkoeppe/s/sage/sage-rebasing/local/lib -Wl,-rpath,/Users/mkoeppe/s/sage/sage-rebasing/local/lib -compatibility_version 4.1.0 -current_version 4.1.0 -o ../lib/libscipsdp.4.1.0.dylib -install_name libscipsdp.4.1.dylib CMakeFiles/libscipsdp.dir/scipsdp/SdpVarmapper.c.o CMakeFiles/libscipsdp.dir/scipsdp/SdpVarfixer.c.o CMakeFiles/libscipsdp.dir/scipsdp/cons_sdp.c.o CMakeFiles/libscipsdp.dir/scipsdp/cons_savedsdpsettings.c.o CMakeFiles/libscipsdp.dir/scipsdp/cons_savesdpsol.c.o CMakeFiles/libscipsdp.dir/scipsdp/relax_sdp.c.o CMakeFiles/libscipsdp.dir/scipsdp/disp_sdpiterations.c.o CMakeFiles/libscipsdp.dir/scipsdp/disp_sdpavgiterations.c.o CMakeFiles/libscipsdp.dir/scipsdp/disp_sdpfastsettings.c.o CMakeFiles/libscipsdp.dir/scipsdp/disp_sdppenalty.c.o CMakeFiles/libscipsdp.dir/scipsdp/disp_sdpunsolved.c.o CMakeFiles/libscipsdp.dir/scipsdp/prop_sdpredcost.c.o CMakeFiles/libscipsdp.dir/scipsdp/branch_sdpmostfrac.c.o CMakeFiles/libscipsdp.dir/scipsdp/branch_sdpmostinf.c.o CMakeFiles/libscipsdp.dir/scipsdp/branch_sdpobjective.c.o CMakeFiles/libscipsdp.dir/scipsdp/branch_sdpinfobjective.c.o CMakeFiles/libscipsdp.dir/scipsdp/heur_sdpfracround.c.o CMakeFiles/libscipsdp.dir/scipsdp/heur_sdpinnerlp.c.o CMakeFiles/libscipsdp.dir/scipsdp/heur_sdpfracdiving.c.o CMakeFiles/libscipsdp.dir/scipsdp/heur_sdprand.c.o CMakeFiles/libscipsdp.dir/scipsdp/prop_sdpsymmetry.c.o CMakeFiles/libscipsdp.dir/scipsdp/prop_sdpobbt.c.o CMakeFiles/libscipsdp.dir/scipsdp/prop_companalcent.c.o CMakeFiles/libscipsdp.dir/scipsdp/reader_cbf.c.o CMakeFiles/libscipsdp.dir/scipsdp/reader_sdpa.c.o CMakeFiles/libscipsdp.dir/scipsdp/sdpsymmetry.c.o CMakeFiles/libscipsdp.dir/scipsdp/table_relaxsdp.c.o CMakeFiles/libscipsdp.dir/scipsdp/table_slater.c.o CMakeFiles/libscipsdp.dir/scipsdp/scipsdpdefplugins.c.o CMakeFiles/libscipsdp.dir/sdpi/sdpi.c.o CMakeFiles/libscipsdp.dir/sdpi/sdpsolchecker.c.o CMakeFiles/libscipsdp.dir/sdpi/solveonevarsdp.c.o CMakeFiles/libscipsdp.dir/sdpi/lapack_interface.c.o CMakeFiles/libscipsdp.dir/sdpi/sdpiclock.c.o CMakeFiles/libscipsdp.dir/scipsdpgithash.c.o CMakeFiles/libscipsdp.dir/sdpi/sdpisolver_none.c.o CMakeFiles/libscipsdp.dir/symmetry/compute_symmetry_bliss.cpp.o  /Users/mkoeppe/s/sage/sage-rebasing/local/lib/libbliss.a /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/liblapack.tbd /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/libblas.tbd -lm 
...
scip_sdp-4.1.0]   "_SCIPverbMessage", referenced from:
[scip_sdp-4.1.0]       _consLockSdp in cons_sdp.c.o
[scip_sdp-4.1.0]       _consExitSdp in cons_sdp.c.o
[scip_sdp-4.1.0]       _consPresolSdp in cons_sdp.c.o
[scip_sdp-4.1.0]       _relaxInitSolSdp in relax_sdp.c.o
[scip_sdp-4.1.0]       _relaxExitSolSdp in relax_sdp.c.o
[scip_sdp-4.1.0]       _SCIPparamChgdSolvesdps in relax_sdp.c.o
[scip_sdp-4.1.0]       _SCIPrelaxSdpComputeAnalyticCenters in relax_sdp.c.o
[scip_sdp-4.1.0]       ...
[scip_sdp-4.1.0]   "_SCIPwarningMessage", referenced from:
[scip_sdp-4.1.0]       _calcRelax in relax_sdp.c.o
[scip_sdp-4.1.0]       _determineSymmetry in prop_sdpsymmetry.c.o
[scip_sdp-4.1.0]       _readerReadSdpa in reader_sdpa.c.o
[scip_sdp-4.1.0] ld: symbol(s) not found for architecture x86_64
[scip_sdp-4.1.0] clang: error: linker command failed with exit code 1 (use -v to see invocation)
[scip_sdp-4.1.0] make[5]: *** [lib/libscipsdp.4.1.0.dylib] Error 1
```
Looks like it forgets to link the scip library

**Comment by mkoeppe** [comment]
<div id="comment:10" align="right">comment:10</div>

Opened https://github.com/scipopt/SCIP-SDP/issues/1

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:11"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0a5647afe1d1f17978b70cb7768819c2e9a2c133"><code>0a5647a</code></a></td><td><code>build/pkgs/scip_sdp/patches: Add macOS build patches</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`8250adb`](https://github.com/sagemath/sagetrac-mirror/commit/8250adb72631c18d7db630b102c715bde64a5d7a)** to **[`0a5647a`](https://github.com/sagemath/sagetrac-mirror/commit/0a5647afe1d1f17978b70cb7768819c2e9a2c133)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`0a5647a`](https://github.com/sagemath/sagetrac-mirror/commit/0a5647afe1d1f17978b70cb7768819c2e9a2c133)** to **[`f90180c`](https://github.com/sagemath/sagetrac-mirror/commit/f90180c75bd30cfec16cbf6fcbcef498e566eee9)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:12"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/f90180c75bd30cfec16cbf6fcbcef498e566eee9"><code>f90180c</code></a></td><td><code>build/pkgs/dsdp: New dummy package</code></td></tr></table>

**Comment by mkoeppe** [comment]
<div id="comment:13" align="right">comment:13</div>

https://github.com/cvxopt/cvxopt-wheels/blob/master/library_builders.sh has an install script for dsdp

**Comment by dimpase** [comment]
<div id="comment:14" align="right">comment:14</div>

DSDP should be a separate Sage package, I think.

**Comment by mkoeppe** [comment]
<div id="comment:15" align="right">comment:15</div>

Yes, but this ancient package has no proper build system. Various ad hoc installation scripts in downstream packages:
- https://github.com/macports/macports-ports/tree/master/math/DSDP
- https://salsa.debian.org/debian/dsdp/-/blob/debian/master/debian/rules
- https://src.fedoraproject.org/rpms/DSDP/blob/rawhide/f/DSDP.Makefile ("Forget it!  I'm writing my own Makefile")
- https://gitweb.gentoo.org/repo/gentoo.git/tree/sci-libs/dsdp

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`f90180c`](https://github.com/sagemath/sagetrac-mirror/commit/f90180c75bd30cfec16cbf6fcbcef498e566eee9)** to **[`f7e21a0`](https://github.com/sagemath/sagetrac-mirror/commit/f7e21a0d0c84429f86d0a1fc1d56dbd1dc2d17d0)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:16"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/98155a5f39e8f4e133a7a07c6c475d04739cc5be"><code>98155a5</code></a></td><td><code>build/pkgs/dsdp: Add installation script</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/4d0267209f417449de4098394f58fd8c5725ca0c"><code>4d02672</code></a></td><td><code>build/pkgs/dsdp/distros/conda.txt: New</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/f7e21a0d0c84429f86d0a1fc1d56dbd1dc2d17d0"><code>f7e21a0</code></a></td><td><code>build/pkgs/scip_sdp: Use upstream patch https://github.com/scipopt/SCIP-SDP/commit/09e79cf75146ca34c637f919d1f4527a63743300, configure blas/lapack</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`f7e21a0`](https://github.com/sagemath/sagetrac-mirror/commit/f7e21a0d0c84429f86d0a1fc1d56dbd1dc2d17d0)** to **[`a90f917`](https://github.com/sagemath/sagetrac-mirror/commit/a90f917e0c2e459aac5961a994f6fb394f9371af)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:17"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a90f917e0c2e459aac5961a994f6fb394f9371af"><code>a90f917</code></a></td><td><code>build/pkgs/dsdp/patches/debian-type-mismatch.patch: New</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:18"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/24dae35ed91fd0d185a459e787384f7fd0459293"><code>24dae35</code></a></td><td><code>build/pkgs/dsdp/distros/repology.txt: New</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`a90f917`](https://github.com/sagemath/sagetrac-mirror/commit/a90f917e0c2e459aac5961a994f6fb394f9371af)** to **[`24dae35`](https://github.com/sagemath/sagetrac-mirror/commit/24dae35ed91fd0d185a459e787384f7fd0459293)**

**Comment by mkoeppe** [comment]
<div id="comment:19" align="right">comment:19</div>

conda-forge wins. it has a cmake build system for DSDP: https://github.com/conda-forge/dsdp-feedstock/tree/main/recipe

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:21"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/45cfd7fca20d6baac88ed73fb1bdc8be68282633"><code>45cfd7f</code></a></td><td><code>build/pkgs/dsdp: Use cmake build script from conda-forge, patch</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`24dae35`](https://github.com/sagemath/sagetrac-mirror/commit/24dae35ed91fd0d185a459e787384f7fd0459293)** to **[`45cfd7f`](https://github.com/sagemath/sagetrac-mirror/commit/45cfd7fca20d6baac88ed73fb1bdc8be68282633)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:22"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/10087901ded3646224ce4b9e81f9fc373525ae9c"><code>1008790</code></a></td><td><code>build/pkgs/scip_sdp/dependencies: Add dsdp</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`45cfd7f`](https://github.com/sagemath/sagetrac-mirror/commit/45cfd7fca20d6baac88ed73fb1bdc8be68282633)** to **[`1008790`](https://github.com/sagemath/sagetrac-mirror/commit/10087901ded3646224ce4b9e81f9fc373525ae9c)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`1008790`](https://github.com/sagemath/sagetrac-mirror/commit/10087901ded3646224ce4b9e81f9fc373525ae9c)** to **[`e2d5f76`](https://github.com/sagemath/sagetrac-mirror/commit/e2d5f76aa2fe839d75deae95a3f734d23cf8ab1c)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:23"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/177bb54277aee2dbe5a8ea807a70ba60413604ba"><code>177bb54</code></a></td><td><code>build/pkgs/scip_sdp/spkg-install.in: Remove unused variable settings</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e2d5f76aa2fe839d75deae95a3f734d23cf8ab1c"><code>e2d5f76</code></a></td><td><code>build/pkgs/scip_sdp/spkg-install.in: Use SDPS=dsdp</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`e2d5f76`](https://github.com/sagemath/sagetrac-mirror/commit/e2d5f76aa2fe839d75deae95a3f734d23cf8ab1c)** to **[`11b0a4a`](https://github.com/sagemath/sagetrac-mirror/commit/11b0a4a8a718237e83087cee718ec9a2f868a251)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:24"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/11b0a4a8a718237e83087cee718ec9a2f868a251"><code>11b0a4a</code></a></td><td><code>build/pkgs/scip_sdp/spkg-check.in: New</code></td></tr></table>

**Comment by mkoeppe** [comment]
Author: **Matthias Koeppe**

**Comment by mkoeppe** [comment]
<div id="comment:26" align="right">comment:26</div>

We could also build cvxopt with dsdp support, but that will be easier after making either dsdp standard or cvxopt optional. Best left for another ticket.

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -5,7 +5,7 @@
 used to be LGPL license, changed to Apache 2.0 in Dec 2022
 
 For full functionality, needs an SDP solver, one of:
-- DSDP - https://repology.org/project/dsdp/versions
+- DSDP - https://repology.org/project/dsdp/versions -- added here as a Sage package
 - SDPA - https://repology.org/project/sdpa/versions
 - MOSEK (non-free)
 
``````

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`11b0a4a`](https://github.com/sagemath/sagetrac-mirror/commit/11b0a4a8a718237e83087cee718ec9a2f868a251)** to **[`805e67d`](https://github.com/sagemath/sagetrac-mirror/commit/805e67d8fd3c5e3e5a19a6532d7837204c171333)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:28"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ebac2fdbc263993bb2a526f5dd41546ba81e3631"><code>ebac2fd</code></a></td><td><code>build/pkgs/papilo/patches: Add https://github.com/scipopt/papilo/commit/e34cfbe0b6d6b4c6e8d5d9305046a00821c81828</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/d26090f4b21f4def3e07b09a55dfa0264137b1dd"><code>d26090f</code></a></td><td><code>build/pkgs/papilo/patches/import_memory_multiprecision.patch: New, back out ineffective patch</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/805e67d8fd3c5e3e5a19a6532d7837204c171333"><code>805e67d</code></a></td><td><code>Merge #34726</code></td></tr></table>

**Comment by mkoeppe** [comment]
Reviewer: **https://github.com/mkoeppe/SCIP-SDP/actions/runs/3660388809**

**Comment by mkoeppe** [comment]
<div id="comment:30" align="right">comment:30</div>

On ubuntu-focal-minimal (https://github.com/mkoeppe/SCIP-SDP/actions/runs/3660388809/jobs/6187479988):

```
[dsdp-5.8]   CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
  [dsdp-5.8]   Please set them or make sure they are set and tested correctly in the CMake files:
  [dsdp-5.8]   BLAS_LIB
  [dsdp-5.8]       linked by target "dsdp" in directory /sage/local/var/tmp/sage/build/dsdp-5.8/src
  [dsdp-5.8]   LAPACK_LIB
  [dsdp-5.8]       linked by target "dsdp" in directory /sage/local/var/tmp/sage/build/dsdp-5.8/src
```

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`805e67d`](https://github.com/sagemath/sagetrac-mirror/commit/805e67d8fd3c5e3e5a19a6532d7837204c171333)** to **[`adc212d`](https://github.com/sagemath/sagetrac-mirror/commit/adc212de1e406055f49995d07878ba656cc2490d)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:31"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/adc212de1e406055f49995d07878ba656cc2490d"><code>adc212d</code></a></td><td><code>build/pkgs/dsdp/dependencies: New</code></td></tr></table>

**Comment by mkoeppe** [comment]
Changed dependencies from **#31329** to **#31329, #34835**

**Comment by mkoeppe** [comment]
Changed reviewer from **https://github.com/mkoeppe/SCIP-SDP/actions/runs/3660388809** to **https://github.com/mkoeppe/SCIP-SDP/actions/runs/3660649526**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`adc212d`](https://github.com/sagemath/sagetrac-mirror/commit/adc212de1e406055f49995d07878ba656cc2490d)** to **[`50b8e2e`](https://github.com/sagemath/sagetrac-mirror/commit/50b8e2ebc14d69a29027ef425abaa1d9c4f0663d)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:34"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e6bb8e6bae03cb7760ccf5674a08cbc59d7e25fd"><code>e6bb8e6</code></a></td><td><code>build/pkgs/openblas/dependencies: Remove python</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/50b8e2ebc14d69a29027ef425abaa1d9c4f0663d"><code>50b8e2e</code></a></td><td><code>Merge #34835</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`50b8e2e`](https://github.com/sagemath/sagetrac-mirror/commit/50b8e2ebc14d69a29027ef425abaa1d9c4f0663d)** to **[`a10dd81`](https://github.com/sagemath/sagetrac-mirror/commit/a10dd819377784c3dfc5390193895b50e1decd92)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:35"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a10dd819377784c3dfc5390193895b50e1decd92"><code>a10dd81</code></a></td><td><code>build/pkgs/dsdp: Use find_package(LAPACK)</code></td></tr></table>

**Comment by mkoeppe** [comment]
<div id="comment:36" align="right">comment:36</div>

debian-bullseye-minimal (https://github.com/mkoeppe/SCIP-SDP/actions/runs/3660649526/jobs/6189250984)

```
  [scip_sdp-git]   /usr/bin/g++  -std=gnu++11 -g -O2 -Wl,-rpath-link,/sage/local/lib -L/sage/local/lib -Wl,-rpath,/sage/local/lib -Wl,-rpath-link,/sage/local/lib -L/sage/local/lib -Wl,-rpath,/sage/local/lib -rdynamic CMakeFiles/scipsdp.dir/scipsdp/main.c.o CMakeFiles/scipsdp.dir/scipsdp/SdpVarmapper.c.o CMakeFiles/scipsdp.dir/scipsdp/SdpVarfixer.c.o CMakeFiles/scipsdp.dir/scipsdp/cons_sdp.c.o CMakeFiles/scipsdp.dir/scipsdp/cons_savedsdpsettings.c.o CMakeFiles/scipsdp.dir/scipsdp/cons_savesdpsol.c.o CMakeFiles/scipsdp.dir/scipsdp/relax_sdp.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdpiterations.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdpavgiterations.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdpfastsettings.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdppenalty.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdpunsolved.c.o CMakeFiles/scipsdp.dir/scipsdp/prop_sdpredcost.c.o CMakeFiles/scipsdp.dir/scipsdp/branch_sdpmostfrac.c.o CMakeFiles/scipsdp.dir/scipsdp/branch_sdpmostinf.c.o CMakeFiles/scipsdp.dir/scipsdp/branch_sdpobjective.c.o CMakeFiles/scipsdp.dir/scipsdp/branch_sdpinfobjective.c.o CMakeFiles/scipsdp.dir/scipsdp/heur_sdpfracround.c.o CMakeFiles/scipsdp.dir/scipsdp/heur_sdpinnerlp.c.o CMakeFiles/scipsdp.dir/scipsdp/heur_sdpfracdiving.c.o CMakeFiles/scipsdp.dir/scipsdp/heur_sdprand.c.o CMakeFiles/scipsdp.dir/scipsdp/prop_sdpsymmetry.c.o CMakeFiles/scipsdp.dir/scipsdp/prop_sdpobbt.c.o CMakeFiles/scipsdp.dir/scipsdp/prop_companalcent.c.o CMakeFiles/scipsdp.dir/scipsdp/reader_cbf.c.o CMakeFiles/scipsdp.dir/scipsdp/reader_sdpa.c.o CMakeFiles/scipsdp.dir/scipsdp/sdpsymmetry.c.o CMakeFiles/scipsdp.dir/scipsdp/table_relaxsdp.c.o CMakeFiles/scipsdp.dir/scipsdp/table_slater.c.o CMakeFiles/scipsdp.dir/scipsdp/scipsdpdefplugins.c.o CMakeFiles/scipsdp.dir/sdpi/sdpi.c.o CMakeFiles/scipsdp.dir/sdpi/sdpsolchecker.c.o CMakeFiles/scipsdp.dir/sdpi/solveonevarsdp.c.o CMakeFiles/scipsdp.dir/sdpi/lapack_interface.c.o CMakeFiles/scipsdp.dir/sdpi/sdpiclock.c.o CMakeFiles/scipsdp.dir/scipsdpgithash.c.o CMakeFiles/scipsdp.dir/sdpi/sdpisolver_dsdp.c.o CMakeFiles/scipsdp.dir/symmetry/compute_symmetry_bliss.cpp.o -o ../bin/scipsdp  -ldsdp -llapack -lblas /sage/local/lib/libscip.so.8.0.2.0 -Wl,-Bstatic -lbliss -Wl,-Bdynamic -L/sage/local/lib -lopenblas -lm 
  [scip_sdp-git]   /usr/bin/ld: cannot find -llapack
  [scip_sdp-git]   /usr/bin/ld: cannot find -lblas
```

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:37"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/8ef2ea4425542b48dcfc7427f90398111824ff98"><code>8ef2ea4</code></a></td><td><code>build/pkgs/scip_sdp/patches/another_blas_change.patch: New</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`a10dd81`](https://github.com/sagemath/sagetrac-mirror/commit/a10dd819377784c3dfc5390193895b50e1decd92)** to **[`8ef2ea4`](https://github.com/sagemath/sagetrac-mirror/commit/8ef2ea4425542b48dcfc7427f90398111824ff98)**

**Comment by mkoeppe** [comment]
<div id="comment:39" align="right">comment:39</div>

https://github.com/scipopt/SCIP-SDP/pull/5

**Comment by mkoeppe** [comment]
<div id="comment:40" align="right">comment:40</div>

`archlinux-latest-standard` (https://github.com/mkoeppe/SCIP-SDP/actions/runs/3660649526/jobs/6190005686)

```
  [scip_sdp-git]   /usr/sbin/g++  -std=gnu++11 -g -O2 -Wl,-rpath-link,/sage/local/lib -L/sage/local/lib -Wl,-rpath,/sage/local/lib -Wl,-rpath-link,/sage/local/lib -L/sage/local/lib -Wl,-rpath,/sage/local/lib -rdynamic CMakeFiles/scipsdp.dir/scipsdp/main.c.o CMakeFiles/scipsdp.dir/scipsdp/SdpVarmapper.c.o CMakeFiles/scipsdp.dir/scipsdp/SdpVarfixer.c.o CMakeFiles/scipsdp.dir/scipsdp/cons_sdp.c.o CMakeFiles/scipsdp.dir/scipsdp/cons_savedsdpsettings.c.o CMakeFiles/scipsdp.dir/scipsdp/cons_savesdpsol.c.o CMakeFiles/scipsdp.dir/scipsdp/relax_sdp.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdpiterations.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdpavgiterations.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdpfastsettings.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdppenalty.c.o CMakeFiles/scipsdp.dir/scipsdp/disp_sdpunsolved.c.o CMakeFiles/scipsdp.dir/scipsdp/prop_sdpredcost.c.o CMakeFiles/scipsdp.dir/scipsdp/branch_sdpmostfrac.c.o CMakeFiles/scipsdp.dir/scipsdp/branch_sdpmostinf.c.o CMakeFiles/scipsdp.dir/scipsdp/branch_sdpobjective.c.o CMakeFiles/scipsdp.dir/scipsdp/branch_sdpinfobjective.c.o CMakeFiles/scipsdp.dir/scipsdp/heur_sdpfracround.c.o CMakeFiles/scipsdp.dir/scipsdp/heur_sdpinnerlp.c.o CMakeFiles/scipsdp.dir/scipsdp/heur_sdpfracdiving.c.o CMakeFiles/scipsdp.dir/scipsdp/heur_sdprand.c.o CMakeFiles/scipsdp.dir/scipsdp/prop_sdpsymmetry.c.o CMakeFiles/scipsdp.dir/scipsdp/prop_sdpobbt.c.o CMakeFiles/scipsdp.dir/scipsdp/prop_companalcent.c.o CMakeFiles/scipsdp.dir/scipsdp/reader_cbf.c.o CMakeFiles/scipsdp.dir/scipsdp/reader_sdpa.c.o CMakeFiles/scipsdp.dir/scipsdp/sdpsymmetry.c.o CMakeFiles/scipsdp.dir/scipsdp/table_relaxsdp.c.o CMakeFiles/scipsdp.dir/scipsdp/table_slater.c.o CMakeFiles/scipsdp.dir/scipsdp/scipsdpdefplugins.c.o CMakeFiles/scipsdp.dir/sdpi/sdpi.c.o CMakeFiles/scipsdp.dir/sdpi/sdpsolchecker.c.o CMakeFiles/scipsdp.dir/sdpi/solveonevarsdp.c.o CMakeFiles/scipsdp.dir/sdpi/lapack_interface.c.o CMakeFiles/scipsdp.dir/sdpi/sdpiclock.c.o CMakeFiles/scipsdp.dir/scipsdpgithash.c.o CMakeFiles/scipsdp.dir/sdpi/sdpisolver_dsdp.c.o CMakeFiles/scipsdp.dir/symmetry/compute_symmetry_bliss.cpp.o -o ../bin/scipsdp  -ldsdp /sage/local/lib/libscip.so.8.0.2.0 -Wl,-Bstatic -lbliss -Wl,-Bdynamic -llapack -lm 
  [scip_sdp-git]   cd /sage/local/var/tmp/sage/build/scip_sdp-git/src/build/src && /usr/bin/cmake -E cmake_symlink_library ../lib/libscipsdp.so.4.1.0 ../lib/libscipsdp.so.4.1 ../lib/libscipsdp.so
  [scip_sdp-git]   [100%] Built target libscipsdp
  [scip_sdp-git]   /usr/sbin/ld: CMakeFiles/scipsdp.dir/sdpi/lapack_interface.c.o: undefined reference to symbol 'dgemm_'
  [scip_sdp-git]   /usr/sbin/ld: /usr/lib/libopenblas.so.3: error adding symbols: DSO missing from command line
  [scip_sdp-git]   collect2: error: ld returned 1 exit status
  [scip_sdp-git]   make[5]: *** [src/CMakeFiles/scipsdp.dir/build.make:695: bin/scipsdp] Error 1
```

**Comment by mkoeppe** [comment]
<div id="comment:41" align="right">comment:41</div>

The failure on `opensuse-tumbleweed-minimal` is tracked in #34764.

The failure on `centos-stream-9-pyhton3.9-minimal` is #34786.

The failure on `ubuntu-trusty-...` comes from #34726.

The failure on `conda-forge` will disappear when #33842 is merged.

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`8ef2ea4`](https://github.com/sagemath/sagetrac-mirror/commit/8ef2ea4425542b48dcfc7427f90398111824ff98)** to **[`b9c9e3d`](https://github.com/sagemath/sagetrac-mirror/commit/b9c9e3d021c5780c99037905c10af377d9424e56)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:42"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/b9c9e3d021c5780c99037905c10af377d9424e56"><code>b9c9e3d</code></a></td><td><code>build/pkgs/scip_sdp/patches/zz_another_blas_change.patch: Link through with blas</code></td></tr></table>

**Comment by mkoeppe** [comment]
<div id="comment:43" align="right">comment:43</div>

This has fixed `archlinux-latest-standard` (comment:40)

**Comment by mkoeppe** [comment]
<div id="comment:44" align="right">comment:44</div>

Ready for review

**Comment by dimpase** [comment]
<div id="comment:45" align="right">comment:45</div>

lgtm

**Comment by dimpase** [comment]
Changed reviewer from **https://github.com/mkoeppe/SCIP-SDP/actions/runs/3660649526** to **Dima Pasechnik**

**Comment by mkoeppe** [comment]
<div id="comment:46" align="right">comment:46</div>

Thank you

**Comment by mkoeppe** [comment]
<div id="comment:47" align="right">comment:47</div>

Pdf docs don't build ([#31329 comment:49](https://github.com/sagemath/sage/issues/31329#comment:49))

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`b9c9e3d`](https://github.com/sagemath/sagetrac-mirror/commit/b9c9e3d021c5780c99037905c10af377d9424e56)** to **[`544d123`](https://github.com/sagemath/sagetrac-mirror/commit/544d123ab56ce2fee80e72c4b4ffeb1fabf305c9)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:48"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/d954f186de7c7fb9b30ad8ab6333a460b146224a"><code>d954f18</code></a></td><td><code>Merge tag '9.8.beta6' into t/34749/package_scip_sdp</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/544d123ab56ce2fee80e72c4b4ffeb1fabf305c9"><code>544d123</code></a></td><td><code>build/pkgs/scip_sdp/SPKG.rst: Remove bad unicode</code></td></tr></table>

**Comment by mkoeppe** [comment]
<div id="comment:49" align="right">comment:49</div>

pdf builds now

**Comment by vbraun** [comment]
Changed branch from **[u/mkoeppe/package_scip_sdp](https://github.com/sagemath/sagetrac-mirror/tree/u/mkoeppe/package_scip_sdp)** to **[`544d123`](https://github.com/sagemath/sagetrac-mirror/commit/544d123ab56ce2fee80e72c4b4ffeb1fabf305c9)**
